airflow/contrib/hooks/bigquery_hook.py

```diff
@@ -487,6 +494,19 @@ def run_load(self,
                 'writeDisposition': write_disposition,
             }
         }
+
+        # if it is a partitioned table ($ is in the table name) add partition load option
+        if '$' in destination_project_dataset_table:
+            if time_partitioning.get('field'):
+                raise AirflowException("Cannot specify field partition and partition name (dataset.table$partition) at the same time")
+            configuration['load']['timePartitioning'] = dict(type='DAY')
+
+        # can specify custom time partitioning options based on a field, or adding expiration
+        if time_partitioning:
+            if not configuration.get('load', {}).get('timePartitioning'):
+                configuration['load']['timePartitioning'] = {}
+            configuration['load']['timePartitioning'].update(time_partitioning)
+
```

Comment:
Hmmm my thought is we should keep `timePartitioning` optional even when table is partitioned (unless there is a good reason to always use timePartitioning for partitioned table). This could simplify the code as well:

```
if time_partitioning:
    if '$' in destination_project_dataset_table and time_partitioning.get('field'):
        raise AirflowException("Cannot specify field partition and partition name (dataset.table$partition) at the same time")
  configuration['load']['timePartitioning'].update(time_partitioning)
```

Reply:
actually scratch what I said, automatic creation of a new partitioned tables makes sense! :D